functional/lenses.md

```diff
@@ -0,0 +1,438 @@
+# Lenses and Prisms
+
+This is a pure functional concept that is not frequently used in Rust, because
+other idioms can achieve the same result and more advanced forms are not
+supported.
+Never the less, exploring the concept may be helpful to understand other
+patterns in Rust APIs, such as
+[visitors](../../patterns/behavioural/visitor.md) and "callbacks".
```

Comment:
Note that I have copied links from other pages that seem to work in style. I don't know why CI is complaining this link is broken anyway.